docs/reference/cluster-config.txt

```diff
@@ -294,6 +296,18 @@ The master supports the following configuration settings:
       -  ``serveraddress`` (optional)
       -  ``email`` (optional)
 
+   -  ``add_capabilities``: The default list of capability strings to
+      pass to the Docker daemon. Ignored by resource managers of type
+      ``kubernetes``. See :ref:`environment.add_capabilities
```

Comment:
curious, how would this work for k8s? devices and caps have to be added to the containers (as far as i see) so I don't think it would be configurable in the current system, unless we propagated these configurations to the pod spec for the user.

Reply:
Basically there's [add/drop caps](https://kubernetes.io/docs/tasks/configure-pod-container/security-context/#set-capabilities-for-a-container) that are pretty straight-forward.  I believe k8s users could specify those things in the pod spec right now on tip-of-master.  I chose to not propagate these settings into the pod spec because a) I don't like multiple ways to configure the same thing, and b) it's easier to add it later if this is the wrong call than to remove it later.

But I'm open to reversing that decision, if anybody feels strongly.

However, [supporting arbitrary devices](https://github.com/kubernetes/kubernetes/issues/7890#issuecomment-766088805) is like holy-shit-hard.  Way beyond the scope of this work.

Reply:
> Basically there's add/drop caps that are pretty straight-forward.

Yeah these are specified on the container, so I didn't know how to specify them in the pod spec, wasn't sure how the merging behavior worked if a container spec is in the pod spec.